src/components/OpenWith/OpenWith.js

```diff
@@ -208,22 +217,85 @@ class OpenWith extends PureComponent<Props, State> {
     }
 
     /**
-     * Renders the Open With element
+     * Click handler when an integration is clicked
+     *
+     * @private
+     * @return {void}
+     */
+    onIntegrationClick(): void {
+        /* no-op */
+    }
+
+    /**
+     * Gets the tooltip text for the OpenWith button
+     *
+     * @private
+     * @return {?string | ?Element}
+     */
+    getTooltip(): ?string | ?Element {
```

Comment:
Should this live in the OpenWithButton component, instead? Its job is to basically examine the props passed in below and choose which text to display.

Reply:
There's a couple of things not passed into the component that this function needs to consider. 
1. If there was an error
2. if there's more than one integration

Are you suggesting to include those and push the logic down to the component?

Reply:
I'm not sure how the dropdown menu is going to play into this architecture, but it looks like this is set up as a container -> component relationship with OpenWithButton at the moment. This file is already doing a lot, so passing those props down and allowing the button to handle rendering logic doesn't seem all that bad to me.

Reply:
the BRUI dropdown accepts a child header item, so I'm expecting

if <=1 integration  <OpenWithButton />
if > 1 integration <OpenWithDropdown> <OpenWithButton /> <OpenWithDropdown>

I think passing an error and a length of integrations is reasonable.